modules/soe_paragraph_cta_list/templates/cta-list-paragraph.html.twig

```diff
@@ -0,0 +1,47 @@
+
+{#
+/**
+ * @file
+ * CTA List Paragraph
+ *
+ * Template for the CTA List paragraph type.
+ *
+ * Available variables:
+ *
+ * {{ header }}
+ * {{ deck }}
+ * {{ links }}
+ *
+ */
+#}
+{%- set attributes = attributes.addClass(['su-cta-list-paragraph', modifier_class]) -%}
+{%- set link_count = links.su_cta_list_links['#items']|length -%}

```

Comment:
Patterns are supposed to be generic to any entity type and should not have references to specific fields in them. That requirement is going to make this approach much harder. The `{{ link }}` variable is synonymous with a layout region and could have just about anything in it. A different approach is going to be needed with either a pre-process or field formatted prior to hitting the template.

Reply:
Per your feedback, I went with the approach which we used here: https://github.com/SU-SWS/jumpstart_ui/blob/8.x-1.x/templates/macros/field-links-to-list.twig

which prevents us from having to reference the field by name, but still lets us keep the logic in the template.